### PR TITLE
Asy branch

### LINKED ISSUE #1: Create Button Component
You are to create a button component for reuse throughout the whole project.

In the `components` folder, create a folder named `ui`, then create the `Button` component inside this folder. Every small component like Button, Text, etc will be inside the `components/ui` folder, while other components will be stored inside the `components` folder directly.

The button will have a width of 100% by default.
The following props are required for the button: variant & className
- variant: the values can either be `solid` or `outline`
- colorScheme: can be `primary`, `danger`

Be sure to ask any question at all.

### PR DESCRIPTION
**Pull request by asyinka**
Button component created

resolves #1

### CODE REVIEW

**Review by asyinka — COMMENTED** [review]
#1 chore: remove margin top from button component

**Comment by cremirdevio** [comment]
Before I go ahead and review your code, I want to point out things we discussed about but you didn't put them to use.

1. Branch Naming: Your branch name is totally unrelatable. I showed you an example of my own branch when we were on call yesterday, I named it `component-badge` or so. The name clearly tells what the branch is created for. We did all this when we learnt git.

> Your branch name should be meaningful, concise and all lowercase. There should be no space like you have, if at all I want to use your branch name, I will call it `asy-branch`, you can see how I separated it with hypen instead of space. And you can see its all lower cased.

> For this your branch, I expected a name like `component-button`, very clear and concise. Other suggestioins are `feat-button`, `ui-button` etc

2. Don't just use `feat` in a commit, I mentioned it yesterday. Just use it once for that new feature or component, then the rest of the time, use `chore`. When you are in doubt, please ask, you are learning, take advantage of it.

3. In your PR, after writing the message, on the last line, you will then add, `resolves #1` depending on the issue id. In your case, you just wrote `#1`, this would do nothing when I accept the PR. You can correct this one now. Just click on edit to do that.

**Comment by asyinka** [comment]
1. Branch Naming: I misinterpreted this part. Regardless, I can rename my git branch, delete this and push again.
2. I used the inbuilt VS Code git push and after pushing to the repositories, the commits were not cleared and had to push again only to see the commit happened with the 'feat' keyword twice.
3. Has been edited and correction made

**Comment by cremirdevio** [comment]
@asyinka  Okay, well understood. Just take note. No need to change the name. Its fine.

**Comment by asyinka** [comment]
update: added style to button component, resolves #1

File: src/components/ui/button.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// import { ReactPropTypes } from "react";
+import PropTypes from "prop-types";
+
+export default function Button({ variant, size, children }) {
+  let bgColor;
+  let btnSize;
+  let className;
```
Comment (cremirdevio) [inline src/components/ui/button.jsx:7]: This line is redundant. It's not needed.

File: src/components/ui/button.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+// import { ReactPropTypes } from "react";
+import PropTypes from "prop-types";
+
+export default function Button({ variant, size, children }) {
```
Comment (cremirdevio) [inline src/components/ui/button.jsx:4]: `className` should be a prop. We don't need children prop on the button.

**Review by cremirdevio — CHANGES_REQUESTED** [review]
Correct those 2 changes and push so I can merge

File: src/components/ui/button.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Button.propTypes = {
+  variant: PropTypes.string,
+  children: PropTypes.node,
+  size: PropTypes.string,
+};
+
+Button.defaultProps = {
+  variant: "solid",
+  children: "default",
+  size: "sm",
```
Comment (cremirdevio) [inline src/components/ui/button.jsx:53]: `size: "default"` as the default

Comment (asyinka) [reply src/components/ui/button.jsx:53]: The stated corrections have been made on the code which resolves #1 .

**Review by cremirdevio — APPROVED** [review]
Good job

**Pull request merged**